include/boost/geometry/views/closeable_view.hpp

```diff
@@ -44,19 +47,23 @@ struct closing_view
 {
     // Keep this explicit, important for nested views/ranges
     explicit inline closing_view(Range& r)
-        : m_range(r)
+        : m_range(boost::ref(r))
     {}
 
     typedef closing_iterator<Range> iterator;
     typedef closing_iterator<Range const> const_iterator;
 
-    inline const_iterator begin() const { return const_iterator(m_range); }
-    inline const_iterator end() const { return const_iterator(m_range, true); }
+    inline const_iterator begin() const { return const_iterator(cref()); }
+    inline const_iterator end() const { return const_iterator(cref(), true); }
+
+    inline iterator begin() { return iterator(ref()); }
+    inline iterator end() { return iterator(ref(), true); }
 
-    inline iterator begin() { return iterator(m_range); }
-    inline iterator end() { return iterator(m_range, true); }
 private :
-    Range& m_range;
+    inline Range& ref() { return m_range; }
+    inline Range const& cref() const { return m_range; }
+
+    boost::reference_wrapper<Range> m_range;
 };
 
```

Comment:
@awulkiew I do understand your concerns., and I understand that you may not like to have `closing_view<>` and `identity_view<>` being copyable. I propose, then, to add a second template parameter to both these views (and inevitably a third template parameter to `closeable_view<>` as well) that would indicate if the view is supposed to be copyable. I can then implement/specialize these views to be or not be copyable, depending on the value of the template parameter. The default value would be `false` (which also means backward compatibility), and in the case of `boundary_view<>` I can simply instantiate `closeable_view<>` with `true`.

I agree that the meaning is slightly different, and indeed we need something more close to our geometries (polygon, multilinestring and multipolygon are nested ranges, i.e., ranges of ranges, and I believe this is what creates the "problem").

I have not found an easier way to implement this. Any suggestions are of course welcome.